### PR TITLE
[Spec] Update spec of code.move

### PR DESCRIPTION
**Pull request by UIZorrot**
### Description

This is a part of the spec work for the Aptos Framework from MoveBit. The changes are:

Add spec as much as possible in code.move;
We tired some way, but still can't bypass the vector::enum loop.

### Test Plan

aptos move prove -f code

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/code.move [hunk 1]
```diff
@@ -145,12 +145,14 @@ module aptos_framework::code {
         let allowed_deps = check_dependencies(addr, &pack);
 
         // Check package against conflicts
+        // ....
         let module_names = get_module_names(&pack);
-        let packages = &mut borrow_global_mut<PackageRegistry>(addr).packages;
-        let len = vector::length(packages);
+        let package_immutable = &borrow_global<PackageRegistry>(addr).packages;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/code.move:150]: @UIZorrot , is this code change necessary? or, can we avoid?

**Review by rahxephon89 — CHANGES_REQUESTED** [review]

File: aptos-move/framework/aptos-framework/sources/code.move (hunk 1 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: this change is suggest by teng, which could Bypass the compiler error

File: aptos-move/framework/aptos-framework/sources/code.spec.move [hunk 2]
```diff
@@ -23,27 +23,32 @@ spec aptos_framework::code {
...
+        // TODO: Can't verify 'vector::enumerate' loop.
+        pragma aborts_if_is_partial;
+        let addr = signer::address_of(owner);
+        modifies global<PackageRegistry>(addr);
+        aborts_if pack.upgrade_policy.policy <= upgrade_policy_arbitrary().policy;
     }
 
     spec publish_package_txn {
-        // TODO: Calls `publish_package`.`
+        // TODO: Calls `publish_package`.`a
```
[the remark this answers is not in the document]
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/code.spec.move:34]: oh its a mistake

File: aptos-move/framework/aptos-framework/sources/code.move (hunk 1 above)
Comment (rahxephon89) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: Yes, this is due to a compiler bug: https://github.com/aptos-labs/aptos-core/issues/8752

Comment (rahxephon89) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: @movekevin @xindingw We need to make the ref immutable before passing it to `enumerate_ref` below to make the prover work. Could you give some feedbacks here? Thanks

Comment (xindingw) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: It makes sense. Can you add a comment to clarify the purpose?

Comment (rahxephon89) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: @UIZorrot Could you add a comment to explain the purpose for this code change?

**Review by alokeutpal — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/code.move (hunk 1 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/code.move:150]: surely

**Pull request merged**